libsolidity/formal/SymbolicVariables.cpp

```diff
@@ -171,27 +180,45 @@ SymbolicMappingVariable::SymbolicMappingVariable(
 	string _uniqueName,
 	EncodingContext& _context
 ):
-	SymbolicVariable(move(_type), move(_uniqueName), _context)
+	SymbolicVariable(_type, move(_type), move(_uniqueName), _context)
 {
 	solAssert(isMapping(m_type->category()), "");
 }
 
 SymbolicArrayVariable::SymbolicArrayVariable(
 	solidity::TypePointer _type,
+	solidity::TypePointer _originalType,
 	string _uniqueName,
 	EncodingContext& _context
 ):
-	SymbolicVariable(move(_type), move(_uniqueName), _context)
+	SymbolicVariable(move(_type), move(_originalType), move(_uniqueName), _context)
 {
 	solAssert(isArray(m_type->category()), "");
 }
 
+Expression SymbolicArrayVariable::currentValueAsType(solidity::TypePointer const& _targetType) const
+{
+	solAssert(_targetType, "");
+	// StringLiterals are encoded as SMT arrays in the generic case,
+	// but they can also be compared/assigned to fixed bytes, in which
+	// case they'd need to be encoded as numbers.
+	if (m_originalType->category() == solidity::Type::Category::StringLiteral)
+		if (_targetType->category() == solidity::Type::Category::FixedBytes)
+		{
+			auto strType = dynamic_cast<StringLiteralType const*>(m_originalType);
```

Comment:
You could just have used this in the outer ``if`` (the ``dynamic_cast`` will return ``nullptr``, if the category is wrong).